File: AFBack/DTOs/FriendDTO.cs
using AFBack.Common.DTOs;

namespace AFBack.DTOs;
// Brukes til å hente om en bruker er venn eller ikke, samt ved sletting og oppretting
public class FriendDTO
{
    public int CurrentUserId { get; set; } // Henter den innloggede brukeren
    public DateTime CreatedAt { get; set; } // Når vennskapet ble laget

    public int UserToFriendUserScore { get; set; } 
    public int FriendUserToUserScore { get; set; }
    
    public UserSummaryDto Friend { get; set; } = null!;
}

File: shared/types/FriendDTO.ts
// Brukes til å hente vennene til hook useFriends.ts som vises på freinds/page.tsx. Henter FriendsDTO.cs i backend

  
export interface FriendDTO {
    currentUserId: string;
    createdAt: string;
    userToFriendUserScore: number;
    friendUserToUserScore: number;
    friend: {
      id: string;
      fullName: string;
      profileImageUrl: string | null;
    };
  }